snapcraft/parts/plugins/python_plugin.py

```diff
@@ -64,3 +65,33 @@ def _get_system_python_interpreter(self) -> Optional[str]:
             confinement,
         )
         return interpreter
+
+    @classmethod
+    def post_prime(cls, step_info: StepInfo) -> None:
+        """Perform Python-specific actions right before packing."""
+        base = step_info.project_base
+
+        if base != "core24":
+            # Only fix pyvenv.cfg on core24 snaps
+            return
+
+        root_path: Path = step_info.prime_dir
+
+        pyvenv = root_path / "pyvenv.cfg"
+        if not pyvenv.is_file():
+            return
+
+        snap_path = Path(f"/snap/{step_info.project_name}/current")
+        new_home = f"home = {snap_path}"
+
+        candidates = (
+            step_info.part_install_dir,
+            step_info.stage_dir,
+        )
+
+        contents = pyvenv.read_text()
+        for candidate in candidates:
+            old_home = f"home = {candidate}"
+            contents = contents.replace(old_home, new_home)
+
+        pyvenv.write_text(contents)
```

Comment:
WDYT about adding a `debug` statement so users are informed that snapcraft is modifying `pyvenv.cfg` after it is primed?

Reply:
ITIAGI (I think it's a good idea)